### PR TITLE
Remove the GitStatus plugin.

### PR DESCRIPTION
**Pull request by grynspan**
Remove the GitStatus plugin.

A recent PR on Swift Package Manager (https://github.com/apple/swift-package-manager/pull/7202) added a `gitInformation` property to the package manifest's `Context` global. This property contains exactly the information we need from the GitStatus plugin. Let's remove it, then!

Adopting the `gitInformation` property requires that we drop remaining support for Swift 5.10. We will continue to minimally support it in a separate Package.swift file (where we simply won't have git repo information at build time.) This PR does not strip out other 5.10 support code such as `XCTestScaffold`.

Resolves rdar://121529847.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
Because this change requires Swift 5.11, I am holding on to it until we drop remaining support for Swift 5.10.

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

File: Package@swift-5.11.swift [hunk 1]
```diff
@@ -0,0 +1,160 @@
+// swift-tools-version: 5.11
+
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
```
Comment (stmontgomery) [inline Package@swift-5.11.swift:6]: 2024, now 🍾

Comment (grynspan) [reply Package@swift-5.11.swift:6]: Yes but the manifest file is from 2023. 😛

**Review by stmontgomery — APPROVED** [review]

**Pull request merged**